### PR TITLE
Update package:vm_service to ^3.0.0

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

File: packages/devtools_server/pubspec.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ dependencies:
   shelf_static: ^0.2.8
   http_multi_server: ^2.2.0
   usage: ^3.4.1
-  vm_service: ^2.3.1
+  vm_service: ^3.0.0
 
 dependency_overrides:
   devtools_shared:
-    path: ../devtools_shared
+    path: ../devtools_shared
```
Comment (kenzieschmoll) [inline packages/devtools_server/pubspec.yaml]: new line at end of file

File: packages/devtools/pubspec.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ description: A suite of web-based performance tooling for Dart and Flutter.
 # package remaining purely as a container for the compiled copy of the devtools
 # app. That ensures that version constraints for the devtools_app do not impact
 # this package.
-version: 0.2.2
+version: 0.2.3
```
Comment (kenzieschmoll) [inline packages/devtools/pubspec.yaml]: please hold off on changing the devtools version numbers throughout this CL. I have a separate [PR](https://github.com/flutter/devtools/pull/1689) that updates the version numbers (we are doing another release this week). Go ahead and leave the changelog updates -  I will handle the merge conflicts on my CL

File: packages/devtools/pubspec.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ environment:
   sdk: '>=2.3.0 <3.0.0'
 
 dependencies:
-  devtools_server: ^0.1.15
+  devtools_server: ^0.1.16
```
Comment (kenzieschmoll) [inline packages/devtools/pubspec.yaml]: 0.1.16 does not exist - I don't know if this matters for pub. 0.2.2 is the latest released version of the server

File: packages/devtools_app/pubspec.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: Main package implementing web-based performance tooling for Dart an
 # When publishing new versions of this package be sure to publish a new version
 # of package:devtools as well. package:devtools contains a compiled snapshot of
 # this package.
-version: 0.2.2
+version: 0.2.3
```
Comment (kenzieschmoll) [inline packages/devtools_app/pubspec.yaml]: please revert this change - I'll bump the version in my release prep cl.

 changelog entries are fine to stay

File: packages/devtools_app/lib/devtools.dart [hunk 5]
```diff
@@ -5,4 +5,4 @@
 /// The DevTools application version.
 // Note: when updating this, please update the corresponding version in the
 // pubspec.
-const String version = '0.2.2';
+const String version = '0.2.3';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/devtools.dart]: please revert this change

File: packages/devtools/pubspec.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ environment:
   sdk: '>=2.3.0 <3.0.0'
 
 dependencies:
-  devtools_server: ^0.1.15
+  devtools_server: ^0.2.2
```
Comment (kenzieschmoll) [inline packages/devtools/pubspec.yaml]: once I release devtools_server 0.2.3 (hopefully today), will I need to bump this to 0.2.3?

Comment (bkonyi) [reply packages/devtools/pubspec.yaml]: I think that's less of a priority as Flutter doesn't depend on it, but it probably wouldn't hurt.

**Comment by kenzieschmoll** [comment]
I was going to publish devtools_server today to keep all the version numbers synced. We need to release a new version of DevTools to disable the survey. See https://github.com/flutter/devtools/pull/1689

**Comment by bkonyi** [comment]
> Why do we need to publish a new version of devtools_server?
> It would be better to not publish a new version devtools_server until we want to release devtools again
> or to update the pubspecs of all 4 packages that we keep tidally locked (on the same version number)
> becase we use dependency overrides for all local development.

Something depends on `devtools_server` in `flutter_tools` which is blocking us from updating `package:vm_service` to 3.0.0 which contains major bug fixes. We can't roll forward without this being updated.

**Comment by devoncarew** [comment]
> Something depends on `devtools_server` in `flutter_tools` which is blocking us from updating `package:vm_service` to 3.0.0 which contains major bug fixes.

Likely `package:dwds`: https://github.com/dart-lang/webdev/blob/master/dwds/pubspec.yaml#L18

**Comment by bkonyi** [comment]
It looks like Travis won't be happy until `devtools_server` is published. @kenzieschmoll is this safe to merge considering you're going to be doing a release shortly?

**Comment by kenzieschmoll** [comment]
nothing in this CL depends on devtools version 0.2.3, so why would devtools_server 0.2.3 need to be published to fix travis?

**Comment by kenzieschmoll** [comment]
Ben I pulled your changes into https://github.com/flutter/devtools/pull/1696, so we don't have to land this on red. The dependency upgrade will go in with that CL.
